stdlib/number.gr

```diff
@@ -439,3 +439,63 @@ export let sin = (radians: Number) => {
 export let cos = (radians: Number) => {
   sin(pi / 2 + radians)
 }
+
+/**
+ * Computes the tangent of a number (in radians) using Chebyshev polynomials.
+ *
+ * @param radians: The input in radians
+ * @returns The computed tangent
+ *
+ * @since v0.5.3
+ */
+export let tan = (radians: Number) => {
+  sin(radians) / cos(radians)
+}
+
+/**
+ * Computes the factorial of a number.
+ *
+ * @param input: The number to compute the factorial of, Must be a Non negative integer
+ * @returns The computed factorial
+ *
+ * @since v0.5.3
+ */
+export let factorial = input => {
+  // Error If It is Negative, or Not an Integer
+  if (sign(input) is -1 || isInteger(input)) {
+    fail "Factorial Expects A Non Negative Integer Input"
```

Comment:
Do we actually want to fail or return an option? @ospencer

Reply:
It should fail. If we really don't want it to fail then we can take Jake's suggestion to define this as the gamma function for non-integers.